### PR TITLE
[Code health] Clean up remote persistence observables

### PR DESCRIPTION
**Pull request by gino-m**
Work towards #699.

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteStorageManager.java [hunk 1]
```diff
@@ -28,8 +29,11 @@
 public interface RemoteStorageManager {
 
   /** Fetches url of a remote file path. */
```
Comment (shobhitagarwal1612) [inline gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteStorageManager.java:31]: Please remove the old comment.

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeaturesCollectionReference.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ public FeatureDocumentReference feature(String id) {
   }
 
   /** Retrieves all features in the project, then streams changes to the remote db incrementally. */
+  @Cold
```
Comment (shobhitagarwal1612) [inline gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeaturesCollectionReference.java:39]: `terminates = false`?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteStorageManager.java (hunk 1 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteStorageManager.java]: Done! Probably some merging mess.

File: gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeaturesCollectionReference.java (hunk 2 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeaturesCollectionReference.java:39]: Good catch!

**Comment by gino-m** [comment]
Thanks @shobhitagarwal1612, PTAL!

**Comment by gino-m** [comment]
> Thanks @shobhitagarwal1612, PTAL!

@shobhitagarwal1612  Fixes some failing checks, please take a look?

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**